drivers/sensor/mcux_tempmon/mcux_tempmon.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2023 Centralp
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_mcux_tempmon
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mcux_tempmon, CONFIG_SENSOR_LOG_LEVEL);
+
+/* OTP Controller Analog Register 1 for calibration values */
+#define OCOTP_ANA1_ADDR             0x401F44E0
+#define OCOTP_ANA1_ROOM_COUNT_SHIFT 20
+#define OCOTP_ANA1_ROOM_COUNT_MASK  (BIT_MASK(12) << OCOTP_ANA1_ROOM_COUNT_SHIFT)
+#define OCOTP_ANA1_HOT_COUNT_SHIFT  8
+#define OCOTP_ANA1_HOT_COUNT_MASK   (BIT_MASK(12) << OCOTP_ANA1_HOT_COUNT_SHIFT)
+#define OCOTP_ANA1_HOT_TEMP_SHIFT   0
+#define OCOTP_ANA1_HOT_TEMP_MASK    (BIT_MASK(8) << OCOTP_ANA1_HOT_TEMP_SHIFT)
+
+#define TEMPMON_ROOM_TEMP 25.0f
+
+struct mcux_tempmon_data {
+	uint8_t hot_temp;
+	uint16_t hot_cnt;
+	uint16_t room_cnt;
+};
+
+struct mcux_tempmon_config {
+	TEMPMON_Type *tempmon;
```

Comment:
can we call this field "base" just to be consistent with the other mcux drivers in the tree?

Reply:
Yeah, sure.